modules/new_serverpod_auth/serverpod_auth_session/serverpod_auth_session_flutter/lib/src/session_manager.dart

```diff
@@ -0,0 +1,155 @@
+  Future<void> init() async {
+    if (_init) {
+      throw RepeatedSessionManagerInitError();
+    }
+    _init = true;
+
+    final sessionKey = await _secureStorage.get(
+      SessionManagerStorageKeys.sessionKeyStorageKey,
+    );
+
+    if (sessionKey != null) {
+      _key = sessionKey;
+
+      final persistedUserId = await _storage.get(
+        SessionManagerStorageKeys.authUserIdStorageKey,
+      );
+      if (persistedUserId == null) {
+        throw const IncompleteSessionManagerStorageException(
+          SessionManagerStorageKeys.authUserIdStorageKey,
+        );
+      }
+
+      final persistedScopeNames = await _storage.get(
+        SessionManagerStorageKeys.scopeNamesStorageKey,
+      );
+      if (persistedScopeNames == null) {
+        throw const IncompleteSessionManagerStorageException(
+          SessionManagerStorageKeys.scopeNamesStorageKey,
+        );
+      }
+
+      _authInfo.value = (
+        authUserId: UuidValue.fromString(persistedUserId),
+        scopeNames:
+            (jsonDecode(persistedScopeNames) as List).cast<String>().toSet(),
+      );
+    }
+  }
+
+  /// Set the current session to a logged-in user described by [authSuccess].
+  Future<void> setLoggedIn(AuthSuccess authSuccess) async {
+    _key = authSuccess.sessionKey;
+
+    _secureStorage.set(
+      SessionManagerStorageKeys.sessionKeyStorageKey,
+      authSuccess.sessionKey,
+    );
+
+    _storage.set(
+      SessionManagerStorageKeys.authUserIdStorageKey,
+      authSuccess.authUserId.uuid,
+    );
+
+    _storage.set(
+      SessionManagerStorageKeys.scopeNamesStorageKey,
+      jsonEncode(authSuccess.scopeNames.toList()),
+    );
+  }
+
+  /// Drop the current user.
+  ///
+  /// This should be called after a `logout` endpoint has been called on the server.
+  Future<void> logout() async {
+    await _secureStorage.set(
+        SessionManagerStorageKeys.sessionKeyStorageKey, null);
+    await _storage.set(SessionManagerStorageKeys.authUserIdStorageKey, null);
+    await _storage.set(SessionManagerStorageKeys.scopeNamesStorageKey, null);
+
+    _key = null;
+    _authInfo.value = null;
+  }
+
+  @override
+  Future<String?> get() async {
+    return _key;
+  }
+
+  @override
+  Future<void> put(String key) {
+    // This is never called by the Serverpod `Client`, and thus no need to implement it.
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<void> remove() {
+    // This is never called by the Serverpod `Client`, and thus no need to implement it.
+    throw UnimplementedError();
+  }
+}
+
+/// Info about the logged-in user.
+typedef AuthInfo = ({UuidValue authUserId, Set<String> scopeNames});
+
+@internal
+abstract class SessionManagerStorageKeys {
+  static const sessionKeyStorageKey = 'serverpod_auth_session.sessionKey';
+  static const authUserIdStorageKey = 'serverpod_auth_session.authUserId';
+  static const scopeNamesStorageKey = 'serverpod_auth_session.scopeNames';
+}
```

Comment:
Perhaps use an enum:
```dart
enum SessionManagerStorageKeys {
  sessionKeyStorageKey('serverpod_auth_session.sessionKey'),
  authUserIdStorageKey('serverpod_auth_session.authUserId'),
  scopeNamesStorageKey('serverpod_auth_session.scopeNames');

  const SessionManagerStorageKeys(this.name);
  final String name;
}
```

Reply:
Would be nice in the test to then iterate over all the known ones easily in the "success case".

Will do.

(On the downside I feel like the access name are already too long, but in another PR we had the discussion that we should not have `@visibleForTesting` fields on the class, as external packages could get it (if they don't use IDE nor lint for it). So I just followed that pattern here again.)